x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -146,6 +193,10 @@ export function getClockSkew(
       // child transaction starts after parent thus no adjustment is needed
       return 0;
     }
+    case 'error':
+      return parentItem.skew;
+    case 'agentMark':
+      return 0;
```

Comment:
Can you move these two up since their behaviour is similar to `span`? And please add comments for them as well